llvm/include/llvm/IR/ConstantRangeList.h

```diff
@@ -0,0 +1,105 @@
+//===- ConstantRangeList.h - A list of constant range -----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Represent a list of signed ConstantRange and do NOT support wrap around the
+// end of the numeric range. Ranges in the list are ordered and no overlapping.
+// Ranges should have the same bitwidth. Each range's lower should be less than
+// its upper. Special lists (take 8-bit as an example):
+//
+// {[0, 0)}     = Empty set
+// {[255, 255)} = Full Set
+//
+// For EmptySet or FullSet, the list size is 1 but it's not allowed to access
+// the range.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_IR_CONSTANTRANGELIST_H
+#define LLVM_IR_CONSTANTRANGELIST_H
+
+#include "llvm/ADT/APInt.h"
+#include "llvm/IR/ConstantRange.h"
+#include <cstddef>
+#include <cstdint>
+
+namespace llvm {
+
+class raw_ostream;
+
+/// This class represents a list of constant ranges.
+class [[nodiscard]] ConstantRangeList {
+  SmallVector<ConstantRange, 2> Ranges;
+
+public:
+  /// Initialize a full or empty set for the specified bit width.
+  explicit ConstantRangeList(uint32_t BitWidth, bool isFullSet);
```

Comment:
if we're only using 64 bit widths, I'd remove the `BitWidth` param, especially since the other constructor seems to assume 64 bit width. if there's a need for more than just 64 in the future, we can add that then

Reply:
Make sense! Removed the `BitWidth` parameter and set to 64 bits by default.